nifi-nar-bundles/nifi-shopify-bundle/nifi-shopify-processors/src/main/java/org/apache/nifi/processors/shopify/GetShopify.java

```diff
@@ -0,0 +1,377 @@
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return RELATIONSHIPS;
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        final StateMap state = getState(context);
+        final boolean isIncremental = context.getProperty(IS_INCREMENTAL).asBoolean();
+        final String initialStartTime = context.getProperty(INITIAL_INCREMENTAL_FILTER).getValue();
+        final Long incrDelayMs = context.getProperty(INCREMENTAL_DELAY).asTimePeriod(TimeUnit.MILLISECONDS);
+
+        String lastExecutionTime = state.get(resourceName);
+        if (lastExecutionTime == null && initialStartTime != null) {
+            lastExecutionTime = initialStartTime;
+        }
+
+        Instant now = getCurrentExecutionTime();
+        if (incrDelayMs != null) {
+            now = now.minus(incrDelayMs, ChronoUnit.MILLIS);
+        }
+        final String currentExecutionTime = now.toString();
+
+        String cursor = null;
+        do {
+            final AtomicInteger objectCountHolder = new AtomicInteger();
+            try (HttpResponseEntity response = shopifyRestService.getShopifyObjects(isIncremental, lastExecutionTime, currentExecutionTime, cursor)) {
+                cursor = getPageCursor(response);
+                if (response.statusCode() == HttpResponseStatus.OK.getCode()) {
+                    FlowFile flowFile = session.create();
+                    flowFile = session.write(flowFile, parseHttpResponse(response, objectCountHolder));
+                    if (objectCountHolder.get() > 0) {
+                        session.transfer(flowFile, REL_SUCCESS);
+                        if (cursor == null && isIncremental) {
+                            final Map<String, String> stateMap = new HashMap<>(state.toMap());
+                            stateMap.put(resourceName, currentExecutionTime);
+                            updateState(context, stateMap);
+                        }
+                    } else {
+                        getLogger().debug("Empty response when requested Shopify resource: [{}]", resourceName);
+                        session.remove(flowFile);
+                    }
+                } else if (response.statusCode() >= 400) {
+                    if (response.statusCode() == TOO_MANY_REQUESTS) {
+                        context.yield();
+                        throw new ProcessException(String.format(
+                                "Rate limit exceeded, yielding before retrying request. HTTP %d error for requested URI [%s]",
+                                response.statusCode(), resourceName));
+                    } else {
+                        context.yield();
+                        getLogger().warn("HTTP {} error for requested Shopify resource [{}]", response.statusCode(),
+                                resourceName);
+                    }
+                }
```

Comment:
Shouldn't we at least log something if the response code is neither 200 nor 400?

Reply:
The 2xx response codes are successful ones. The 1xx informational and the 3xx are redirects could be handled as error in this case. What do you think of changing the error condition to 200 <= resp code < 300?